From 47e0f023746b553b6da0e108b56876c8cb6fe2a5 Mon Sep 17 00:00:00 2001
From: surfacebot <surfacebot@users.noreply.github.com>
Date: Sat, 27 Dec 2025 20:05:35 -0800
Subject: [PATCH] Add Surface Laptop Studio 2 to PCI quirks

---
 drivers/pci/quirks.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index ce4c0da99537..a9f73f56426d 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -6360,6 +6360,13 @@ static const struct dmi_system_id no_shutdown_dmi_table[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "Surface Laptop 5"),
 		},
 	},
+	{
+		.ident = "Microsoft Surface Laptop Studio 2"
+	  	.matches = {
+    		DMI_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
+    		DMI_MATCH(DMI_PRODUCT_NAME, "Surface Laptop Studio 2"),
+  		},
+	},
 	{}
 };
 
@@ -6377,3 +6384,5 @@ DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x461f, quirk_no_shutdown);  // Thu
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x462f, quirk_no_shutdown);  // Thunderbolt 4 PCI Express Root Port
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x466d, quirk_no_shutdown);  // Thunderbolt 4 NHI
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x46a8, quirk_no_shutdown);  // GPU
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa71e, quirk_no_shutdown);  // Thunderbolt 4 USB Controller
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa73e, quirk_no_shutdown);  // Thunderbolt 4 NHI
-- 
2.52.0

